File: week6_Python/problem_set_6/sentimental-readability/readability.py
# TODO

# Recreation of readability.c in problem set 2
from cs50 import get_string


def main():
    # Get user text input
    text = get_string("Text: ")

    letters = 0
    words = 0
    sentences = 0

    # Count number of letters, words, and sentences
    for char in text:
        if char.isalpha():
            letters += 1
        if char == " ":
            words += 1
        if char == "." or char == "?" or char == "!":
            sentences += 1
    words += 1

    # Determine Grade
    lettersper100words = letters / words * 100
    sentencesper100words = sentences / words * 100
    grade = 0.0588 * lettersper100words - 0.296 * sentencesper100words - 15.8

    # Print out the grade
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(grade)}")


main()

File: cs50x/week2_Arrays/problem_set_2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Gets input text
    string text = get_string("Text: ");

    // Counts letters, words and sentences with functions
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Determine letters per 100 words, sentences per 100 words, and grade
    float lettersper100words = (float) letters / words * 100;
    float sentencesper100words = (float) sentences / words * 100;
    float grade = 0.0588 * lettersper100words - 0.296 * sentencesper100words - 15.8;

    // Prints out grades depending on the ranges of score
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(grade));
    }
}

int count_letters(string text)
{
    int letters = 0;
    // Loop through each character, add 1 to letters if its an alphabet
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 0;
    // Loop through each character, add 1 to words if its a space
    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    // Add another 1 to words because 2 words are between 1 space
    words++;
    return words;
}

int count_sentences(string text)
{
    // Loop through each character, add 1 to sentences if its a '.' or '!' or '?'
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

